File: DataSets/DataSet-Old/LeetCodeTasks - Testing/Bad/96_unique-binary-search-trees/solutions/cpp/solution.cpp
#include <vector>



class Solution {

public:

    int numTrees(int n) {

        std::vector<int> dp(n + 1);

        dp[0] = 1;
        

        for (int i = 2; i <= n; ++i) {

            for (int j = 1; j <= i; ++j) {

                dp[i] += dp[j] * dp[i - j];

            }

        }

        

        return dp[n];

    }

};

File: DataSets/DataSet-Old/LeetCodeTasks - Testing/Bad/96_unique-binary-search-trees/solutions/java/solution.java
class Solution {

    public int numTrees(int n) {

        long[] catalan = new long[n + 1];

        catalan[0] = catalan[1] = 1;

        for (int i = 1; i <= n; i++) {

            for (int j = 0; j <= i; j++) {

                catalan[i] += catalan[j] * catalan[i - j - 1];

            }

        }

        return (int) catalan[n];

    }

}

